File: wlxx-services/wlxx-system/wlxx-system-provider/src/main/java/com/xzg/wlxx/system/service/UserService.java
package com.xzg.wlxx.system.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.xzg.wlxx.system.client.entity.dto.RegisterUserDto;
import com.xzg.wlxx.system.client.entity.po.UserPo;

/**
 * @author XiaoZG
 */
public interface UserService extends IService<UserPo> {

    UserPo findByUsername(String username);

    boolean register(RegisterUserDto dto);
}

File: wlxx-services/wlxx-system/wlxx-system-provider/src/main/kotlin/com/xzg/wlxx/system/service/UserService.kt
package com.xzg.wlxx.system.service

import com.baomidou.mybatisplus.extension.service.IService
import com.xzg.wlxx.system.client.entity.dto.RegisterUserDto
import com.xzg.wlxx.system.client.entity.po.UserPo

/**
 * @author XiaoZG
 */
interface UserService : IService<UserPo?> {
    fun findByUsername(username: String?): UserPo?
    fun register(dto: RegisterUserDto): Boolean
}
